pkg/apiserver/resthandler.go

```diff
@@ -676,6 +676,13 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 			}
 		}
 
+		canDelete := func() bool {
```

Comment:
I don't think this is the worst solution, but it unfortunately doesn't handle everything we need in the long run here.  The long term direction I had been envisioning for storage here, to collapse all the requirements resthandler has (patch, field level auth, retry of admission) was to move the signature of update from:

    func Update(ctx api.Context, obj runtime.Object) ...

to

    func Update(ctx api.Context, func(ctx api.Context, old runtime.Object) (new runtime.Object, error)) ...

This then makes it the responsibility of the rest handler to define the "logical mutation" (patch, update, admission on the mutation, generic preconditions) and the responsibility of the store to provide the old object, even in an atomic update loop.

I know this is a larger change, but we need to be moving into this direction for a lot of reasons.